csharp/ql/lib/semmle/code/csharp/ExprOrStmtParent.qll

```diff
@@ -67,17 +65,46 @@ private ControlFlowElement getBody(Callable c) {
   result = getStatementBody(c)
 }
 
+pragma[nomagic]
+private Location getASourceLocation(Element e) {
+  result = e.getALocation().(SourceLocation) and
+  not exists(e.getALocation().(SourceLocation).getMappedLocation())
```

Comment:
It looks like the return type of `getMappedLocation()` can be changed to `SourceLocation` (the `result` of that predicate is found in `locations_mapped` which has type `locations_default`). If that is the case the return type of `getASourceLocation` can then also be changed to `SourceLocation`.